src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphAuthDAO.scala

```diff
@@ -68,6 +68,11 @@ class GraphAuthDAO extends AuthDAO with GraphDAO {
   }
 
   override def getMaximumAccessLevel(userSubjectId: String, workspaceId: String, txn: RawlsTransaction): WorkspaceAccessLevel = {
+    val workspaces = listWorkspaces(userSubjectId, txn)
+    workspaces.find(_.workspaceId == workspaceId).map(_.accessLevel).getOrElse(WorkspaceAccessLevels.NoAccess)
```

Comment:
you're still doing all the work to build permissions lists for a bunch of workspaces you don't care about